src/Microsoft.DotNet.Build.Tasks/Targets/publishtest.targets

```diff
@@ -15,11 +15,33 @@
   </Target>
 
   <PropertyGroup>
-    <PrepareForRunDependsOn Condition="'$(IsTestProject)'=='true'">$(PrepareForRunDependsOn);CopyTestToTestDirectory</PrepareForRunDependsOn>
+    <PrepareForRunDependsOn Condition="'$(IsTestProject)'=='true'">$(PrepareForRunDependsOn);CopyTestToTestDirectory;CreateTestNuGetPackage</PrepareForRunDependsOn>
     <TestArchitecture Condition="'$(TestArchitecture)' == ''">x64</TestArchitecture>
     <TestNugetRuntimeId Condition="'$(TestNugetRuntimeId)' == ''">win7-$(TestArchitecture)</TestNugetRuntimeId>
   </PropertyGroup>
 
+  <Target Name="CreateTestNuGetPackage"
+          DependsOnTargets="CopyTestToTestDirectory"
+          Condition="'$(GenerateTestPackages)'=='true'">
+    <PropertyGroup>
+      <TestPackageContent>@(_TestCopyLocalByFileNameWithoutDuplicates->'%(SourcePath)')</TestPackageContent>
+      <TestPackageBuild Condition="'$(TestPackageBuild)'==''">$(BuildNumberMajor)</TestPackageBuild>
+      <TestPackageBuild Condition="'$(TestPackageBuild)'==''">000000</TestPackageBuild>
+      <TestPackageVersion Condition="'$(TestPackageVersion)'==''">1.0.0-prerelease-$(TestPackageBuild)</TestPackageVersion>
+    </PropertyGroup>
+    <MSBuild Projects="$(MSBuildThisFileDirectory)testpackage.nuproj"
+      Properties="
+      IntermediateOutputPath=$(IntermediateOutputPath);
+      OutputPath=$(OutputPath);
+      NuProjPropsPath=$(NuProjDir)NuProj.props;
```

Comment:
Updated.  I added condition guards on NuProjDir and NuProjVersion so if a consuming repo takes a different version, they can override these values.